graph/path/yen_ksp_test.go

```diff
@@ -5,32 +5,81 @@
 package path
 
 import (
+	"math"
+	"reflect"
 	"testing"
 
-	"gonum.org/v1/gonum/graph/path/internal/testgraphs"
 	"gonum.org/v1/gonum/graph"
+	"gonum.org/v1/gonum/graph/simple"
 )
 
 
+var YenShortestPathTests = []struct {
+	Name              string
+	Graph             func() graph.WeightedEdgeAdder
+	Edges             []simple.WeightedEdge
+
+	Query         simple.Edge
+	K int
+	WantPaths     [][]int64
+}{
+	// Positive weighted graphs.
+	{
+		Name:  "simple graph",
+		Graph: func() graph.WeightedEdgeAdder { return simple.NewWeightedDirectedGraph(0, math.Inf(1)) },
+		Edges: []simple.WeightedEdge{
+			{F: simple.Node(0), T: simple.Node(1), W: 3},
+			{F: simple.Node(0), T: simple.Node(2), W: 2},
+			{F: simple.Node(2), T: simple.Node(1), W: 1},
+			{F: simple.Node(1), T: simple.Node(3), W: 4},
+			{F: simple.Node(2), T: simple.Node(3), W: 2},
+			{F: simple.Node(2), T: simple.Node(4), W: 3},
+			{F: simple.Node(3), T: simple.Node(4), W: 2},
+			{F: simple.Node(3), T: simple.Node(5), W: 1},
+			{F: simple.Node(4), T: simple.Node(5), W: 2},
+		},
+		Query:  simple.Edge{F: simple.Node(0), T: simple.Node(5)},
+		K: 3,
+		WantPaths: [][]int64{
+			{0, 2, 3, 5},
+			{0, 2, 4, 5},
+			{0, 1, 3, 5},
+		},	
+	},
+}
+
+func toIntPath(nodePaths [][]graph.Node) [][]int64 {
+	paths := make([][]int64, 0)
```

Comment:
If you do this, it could be `var paths [][]int64`, but since you know the length of the input, you can make it the right length.

```
func pathIDs(paths [][]graph.Node) [][]int64 {
	ids := make([][]int64, len(paths)
	for i, p := range paths {
		ids[i] = make([]int64, len(p))
		for j, n := range p {
			ids[i][j] = n.ID()
		}
	}
	return ids
}
```

Reply:
Please use this code.